src/json-observe.ts

```diff
@@ -0,0 +1,233 @@
+            return Reflect.set(target, key, receiver);
+          }
+          if (target.hasOwnProperty(key)) { //exists
+            if (typeof target[key] === 'undefined') {
+              if (Array.isArray(target)) {
+                instance.defaultCallback({ op: 'replace', path: distPath, value: receiver });
+              }
+              //when `undefined` property is set to something, should generate add (undefined in JSON.stringified to no value)
+              else { //object              
+                instance.defaultCallback({ op: 'add', path: distPath, value: receiver });
+              }
+              return Reflect.set(target, key, receiver);
+            }
+            else {
+              instance.defaultCallback({ op: 'replace', path: distPath, value: receiver });
+              return Reflect.set(target, key, receiver);
+            }
+          }
+          else {
+            instance.defaultCallback({ op: 'add', path: distPath, value: receiver });
+            return Reflect.set(target, key, receiver);
+          }
+        },
+        deleteProperty: function (target, key) {
+          if (typeof target[key] !== 'undefined') {
+            instance.defaultCallback({ op: 'remove', path: path + '/' + JsonObserver.escapePathComponent(key.toString()) });
+          }
+          // else {
+          return Reflect.deleteProperty(target, key);
+        }
+      });
+    return proxy;
+  }
+
+  //grab tree's leaves one by one, encapsulate them into a proxy and return
+  private _proxifyObjectTreeRecursively(root: any, path): any {
+    for (var key in root) {
+      if (root.hasOwnProperty(key)) {
+        if (typeof root[key] === 'object') {
+          var distPath = path + '/' + JsonObserver.escapePathComponent(key);
+          root[key] = this.generateProxyAtPath(root[key], distPath);
+          this._proxifyObjectTreeRecursively(root[key], distPath);
+        }
+      }
+    }
+    return this.generateProxyAtPath(root, "");
+  }
+
+  //this function is for aesthetic purposes
+  private proxifyObjectTree(root: any): any {
+
+    /*
+    while proxyifying object tree,
+    the proxyifying operation itself is being
+    recorded, which in an unwanted behavior,
+    that's why we disable recording through this
+    initial process;
+    */
+
+    this.switchObserverOff();
+    
+    var proxifiedObject = this._proxifyObjectTreeRecursively(root, "");
+
+    /* OK you can record now */
+    this.switchObserverOn();
+
+    return proxifiedObject;
+  }
+
+  constructor(root) {
+
+    this.originalObject = root;
+    this.cachedProxy = null;
+    this.isRecording = false;
+    this.userCallback;
+    var sender = this;
+    /* 
+    instead of setting a boolean to
+    be checked every time an operation
+    is being recorded, I'm removing the 
+    whole callback when it's disabled.
+    Because we are only disabling the callback
+    once in the beginning, no need to `if` every time
+    */
```

Comment:
We don't need to compare, because before, we used to check the condition with every omitted patch. But now, the whole callback is being changed. So this is 100% faster.